lib/cluster-providers/mng-cluster-provider.ts

```diff
@@ -115,16 +125,33 @@ export class MngClusterProvider implements ClusterProvider {
         const maxSize = this.props.maxSize ?? valueFromContext(scope, constants.MAX_SIZE_KEY, constants.DEFAULT_NG_MAXSIZE);
         const desiredSize = this.props.desiredSize ?? valueFromContext(scope, constants.DESIRED_SIZE_KEY, minSize);
 
+        // Create launch template if custom AMI is provided.
+        const lt = new LaunchTemplate(scope, `${id}-lt`, {
+            machineImage: this.props.customAmi,
+            userData: this.props.userData,
+        });
+
         // Create a managed node group.
-        const mng = cluster.addNodegroupCapacity(id + "-ng", {
-            amiType,
+        const commonNodegroupProps = {
             capacityType,
             instanceTypes,
             minSize,
             maxSize,
             desiredSize,
-            releaseVersion
-        });
+        };
+        const nodegroupProps = this.props.customAmi ? {
+            ...commonNodegroupProps,
+            launchTemplateSpec: {
```

Comment:
We should do this only if custom AMI was provided by the user and omit otherwise.

Reply:
I couldn't figure a good solution on this, but I have tried another way of doing this using `if`, and updated the code. Please kindly review again. Thanks :)

Reply:
There is a way to make it more elegant, but the if statement will do for now.